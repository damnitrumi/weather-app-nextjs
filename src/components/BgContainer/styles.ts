import styled, { css } from "styled-components";
import { BgContainerProps } from ".";

export const Wrapper = styled.article<Pick<BgContainerProps, "bgSelection">>`
  ${({ bgSelection }) => css`
    display: flex;
    flex-direction: column;
    background-image: url(${bgSelection});
    background-size: cover;
    background-repeat: no-repeat;
    background-position: center;
    width: 100%;
    height: 100%;
    margin: 0 auto;

    @media(min-width: 640px){
      flex-direction: row;
    }

    @media(min-width: 768px){
      aspect-ratio: 16/9;
      height: auto;
      max-width: 1000px;
      border-radius: 10px;
      overflow: hidden;

    }
  `}
`;
