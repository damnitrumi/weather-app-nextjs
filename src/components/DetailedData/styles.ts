import styled, { css } from "styled-components";

export const Wrapper = styled.section`
  ${() => css`
    padding:20px;
    min-width: 200px;
    backdrop-filter: blur(10px) saturate(180%);
    -webkit-backdrop-filter: blur(10px) saturate(180%);

    @media (min-width: 640px){
      flex-grow: 2;
    }

    @media (min-width: 1024px){
      padding: 40px;
    }
  `}
`;
