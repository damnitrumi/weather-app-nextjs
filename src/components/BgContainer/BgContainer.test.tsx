import { screen } from "@testing-library/react";
import { renderTheme } from "../../styles/render-theme";
import { BgContainer, BgContainerProps } from ".";

const props: BgContainerProps = {
  bgSelection: "clear",
  children: <h1>Oi</h1>,
};

describe("<BgContainer />", () => {
  it("should render bg container and its children", () => {
    renderTheme(<BgContainer {...props} />);

    expect(screen.getByRole("article")).toBeInTheDocument();
    expect(screen.getByRole("article")).toHaveStyle(
      "background-image: url(/assets/images/clear.jpg)",
    );
    expect(screen.getByRole("heading", { name: "Oi" })).toBeInTheDocument();
  });

  it("should match snapshot", () => {
    renderTheme(<BgContainer {...props} />);
    expect(screen.getByRole("article")).toMatchSnapshot();
  });
});
