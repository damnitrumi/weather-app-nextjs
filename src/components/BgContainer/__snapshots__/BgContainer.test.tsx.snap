// Vitest Snapshot v1

exports[`<BgContainer /> > should match snapshot 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  background-image: url(/assets/images/clear.jpg);
  background-size: cover;
  background-repeat: no-repeat;
  background-position: center;
  width: 100%;
  height: 100%;
  margin: 0 auto;
}

@media (min-width:640px) {
  .c0 {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
  }
}

@media (min-width:768px) {
  .c0 {
    aspect-ratio: 16/9;
    height: auto;
    max-width: 1000px;
    border-radius: 10px;
    overflow: hidden;
  }
}

<article
  class="c0"
>
  <h1>
    Oi
  </h1>
</article>
`;
